src/app/components/pages/Login/index.tsx

```diff
@@ -1,44 +1,72 @@
+
 export interface LoginProps {
   setActivePage: React.Dispatch<React.SetStateAction<PAGES>>;
 }
+
+function generateRandomToken() {
+  const randomValues = new Uint8Array(128);
+  window.crypto.getRandomValues(randomValues);
+
+  // Convert the Uint8Array to a base64 string
+  const base64String = btoa(String.fromCharCode.apply(null, randomValues));
+
+  return base64String;
+}
+
 const Login = function (props: LoginProps) {
   const {} = props;
   const [token, setToken] = useState('');
-  return (
-    <div className="login">
-      <form
-        onSubmit={async (e) => {
-          e.preventDefault();
-          const res = await fetch(API_URL + 'users/temp_token_login?token=' + token);
+
+  useEffect(() => {
+    if (token) fetchUserInfo();
+  }, [token]);
+
+  const fetchUserInfo = () => {
+    const intervalId = setInterval(async () => {
+      try {
+        const res = await fetch(`${API_URL}users/temp_token_login?token=${token}`);
+        if (res.status == 200) {
+          clearInterval(intervalId);
           const userInfo = await res.json();
           parent.postMessage({ pluginMessage: { type: 'store-user-info', userInfo: JSON.stringify(userInfo) } }, '*');
-        }}
-      >
-        <h3>Login page</h3>
-        <input
-          value={token}
-          onChange={(e) => {
-            setToken(e.target.value);
-          }}
-          required
-          placeholder="Huehive login token"
-        />
-        <button type="submit">Submit </button>
-        <div >
-          <p className='login-text'>
-            Copy login token from{' '}
-            <a href="https://huehive.co/users/figma_token" target="_blank">
-              HueHive
+        }
+      } catch {}
+    }, 4000);
+  };
+
+  const onLoginButtonPress = () => {
+    const generatedToken = generateRandomToken();
+    window.open(`${API_URL}users/figma_token?token=${generatedToken}`);
+    setToken(generatedToken);
+  };
+
+  return (
+    <div className="login-container">
+      <img src={background} className="background" />
+      <div className="login-outer">
+        {token == '' ? (
+          <div className="login">
+            <h1>HueHive</h1>
+            <h2>Welcomes You!</h2>
+            <p className="intro">
+              Just a few steps before you use the power of ChatGPT to generate some awesome colour schemes for your app.
+            </p>
+            <button onClick={onLoginButtonPress}>Login</button>
+          </div>
+        ) : (
+          <div className="login-loading">
+            <p>Waiting to login...</p>
+            <a href={`${API_URL}users/figma_token?token=${token}`} target="_blank">
+              Page didn't open? click here
             </a>
-          </p>
-          <p className='sign-up-text'>
-            <a href="https://huehive.co/users/sign_up" target="_blank">Sign up </a><span>if you don't have an account on HueHive</span>
-          </p>
-        </div>
-      </form>
+          </div>
+        )}
+      </div>
```

Comment:
should we keep signup link also as most of the users will not have account in huehive and might get confused that only login option is available? We can keep it smaller and at the bottom.

Reply:
That would work, but you will need to support the addition of token to that url as well, else the user will have to create account, then login again.

Reply:
yeah good idea. I have added `intent=signup` param in the figma token api so that it redirects to signup page if user is not logged in
Example:
```
http://localhost:3000/users/figma_token?token=uuid_token&intent=signup
```
We can add a signup button with adding extra param for intent